prow/plugins/projectmanager/projectmanager.go

```diff
@@ -108,61 +115,102 @@ type githubClient interface {
 	CreateProjectCard(columnID int, projectCard github.ProjectCard) (*github.ProjectCard, error)
 }
 
-func handlePR(gc githubClient, projectManager plugins.ProjectManager, log *logrus.Entry, pe github.PullRequestEvent) error {
-	// Only respond to label add or issue/PR open events
-	if pe.Action != github.PullRequestActionOpened &&
-		pe.Action != github.PullRequestActionReopened &&
-		pe.Action != github.PullRequestActionLabeled {
+type eventData struct {
+	id     int
+	number int
+	isPR   bool
+	org    string
+	repo   string
+	state  string
+	labels []github.Label
+	remove bool
+}
+
+type DuplicateCard struct {
+	projectName string
+	issueURL    string
+}
+
+func (m *DuplicateCard) Error() string {
+	return fmt.Sprintf(issueAlreadyInProject, m.issueURL, m.projectName)
+}
+
+func handleIssueRequest(pc plugins.Agent, ie github.IssueEvent) error {
+	if !handleIssueActions[ie.Action] {
+		return nil
+	}
+	eventData := eventData{ie.Issue.ID, ie.Issue.Number, false, ie.Repo.Owner.Login, ie.Repo.Name, ie.Issue.State, ie.Issue.Labels, false}
+	if ie.Action == github.IssueActionUnlabeled {
+		eventData.remove = true
+	}
+
+	return handle(pc.GitHubClient, pc.PluginConfig.ProjectManager, pc.Logger, eventData)
+}
+
+func handlePullRequest(pc plugins.Agent, pe github.PullRequestEvent) error {
+	if !handlePRActions[pe.Action] {
 		return nil
 	}
+	eventData := eventData{pe.PullRequest.ID, pe.Number, true, pe.Repo.Owner.Login, pe.Repo.Name, pe.PullRequest.State, []github.Label{}, false}
+	if pe.Action == github.PullRequestActionUnlabeled {
+		eventData.remove = true
+	}
+
+	return handle(pc.GitHubClient, pc.PluginConfig.ProjectManager, pc.Logger, eventData)
+}
+
+func handle(gc githubClient, projectManager plugins.ProjectManager, log *logrus.Entry, e eventData) error {
+
 	// Get any ManagedProjects that match this PR
-	matchedColumnIDs := getMatchingColumnIDs(gc, projectManager.OrgRepos, pe, log)
+	matchedColumnIDs := getMatchingColumnIDs(gc, projectManager.OrgRepos, e, log)
 
 	// For each ManagedColumn that matches this PR, add this PR to that Project Column
 	// All the matchedColumnID are valid column ids and the checked to see if the project card
 	// we are adding is not already part of the project and thus avoiding duplication.
 	for _, matchedColumnID := range matchedColumnIDs {
-		err := addPRToColumn(gc, matchedColumnID, pe)
+		err := addPRToColumn(gc, matchedColumnID, e)
 		if err != nil {
 			log.WithError(err).WithFields(logrus.Fields{
-				"pullRequestNumber": pe.Number,
-				"matchedColumnID":   matchedColumnID,
+				"issueNumber":     e.number,
```

Comment:
The plugin agent already exposed the number of the issue or PR in a log field so you don't need to include this. (It is exposed as the `pr` field for both PRs and issues.) https://github.com/kubernetes/test-infra/blob/7780df2a12c931233778d19d7685228ff52de7fe/prow/github/types.go#L32-L34

Reply:
Fixed.